sessiontxn/staleread/provider.go

```diff
@@ -86,10 +89,37 @@ func (p *StalenessTxnContextProvider) OnInitialize(ctx context.Context, tp sessi
 }
 
 // OnStmtStart is the hook that should be called when a new statement started
-func (p *StalenessTxnContextProvider) OnStmtStart(_ context.Context) error {
+func (p *StalenessTxnContextProvider) OnStmtStart(ctx context.Context) error {
+	p.ctx = ctx
```

Comment:
`p.ctx` should also be assigned when `OnInitialize` and `OnStmtRetry` even though not all of them will be called. It  can reduce extra assumptions.

Reply:
Done.